File: CPP/FilmBioskop.cpp
using namespace std;    // pakai namespace std (tanpa prefiks std::)

class FilmBioskop : public Film {  // deklarasi class (level 2), turunan dari Film
private:
    int durasiMenit;      // atribut 1: durasi film (menit)
    vector<string> genre; // atribut 2: daftar genre (bisa lebih dari 1)
    string sutradara;     // atribut 3: sutradara film

public:
    // FilmBioskop: Konstruktor untuk menyimpan durasi, genre, dan sutradara.
    FilmBioskop(int id, string judul, int harga, int durasiMenit,vector<string> genre, string sutradara) : Film(id, judul, harga) {
        this->durasiMenit = durasiMenit;  // simpan durasi film (menit)
        this->genre       = genre;        // simpan daftar genre
        this->sutradara   = sutradara;    // simpan nama sutradara
    }

    // ===== getter =====
    // getDurasiMenit: Getter untuk mengambil atribut durasiMenit.
    int getDurasiMenit()      { return durasiMenit; }  // getter durasi film (menit)
    // getGenre: Getter untuk mengambil daftar genre.
    vector<string> getGenre() { return genre; }        // getter daftar genre
    // getSutradara: Getter untuk mengambil atribut sutradara.
    string getSutradara()     { return sutradara; }    // getter nama sutradara

    // ===== setter =====
    // setDurasiMenit: Setter untuk mengubah atribut durasiMenit.
    void setDurasiMenit(int durasiMenit) { this->durasiMenit = durasiMenit; }  // setter durasi film (menit)
    // setGenre: Setter untuk mengubah daftar genre.
    void setGenre(vector<string> genre)  { this->genre = genre; }              // setter daftar genre
    // setSutradara: Setter untuk mengubah atribut sutradara.
    void setSutradara(string sutradara)  { this->sutradara = sutradara; }      // setter nama sutradara

    // getGenreText: Gabungkan daftar genre menjadi satu string dipisah koma.
    string getGenreText() {
        string hasil;                                // penampung genre yang sudah digabung
        for (size_t i = 0; i < genre.size(); i++) {  // telusuri semua genre pada daftar
            if (i > 0) hasil += ", ";                // gunakan koma sebagai pemisah antar genre
            hasil += genre[i];                       // tambahkan genre saat ini ke hasil
        }
        return hasil;                                // kembalikan semua genre dalam satu string
    }

    // getData: Mengambil data Film ditambah atribut bioskop (6 kolom).
    vector<string> getData() {                        // method getData milik FilmBioskop (menutup milik Film)
        vector<string> data = Film::getData();        // ambil 3 kolom dasar dari class Film
        int jam  = durasiMenit / 60;                  // hitung jumlah jam (pembagian integer)
        int sisa = durasiMenit % 60;                  // hitung sisa menit sesudah dibagi jam
        string durasi;
        if (jam > 0) {                                   // jika durasi minimal satu jam
            durasi = to_string(jam) + " jam";            // awali dengan jumlah jam
            if (sisa > 0) durasi += " " + to_string(sisa) + " menit";  // tambahkan sisa menit
        } else {
            durasi = to_string(durasiMenit) + " menit";  // di bawah 1 jam: tampilkan menit saja
        }
        data.push_back(durasi);                          // tambah kolom durasi (format jam/menit)
        data.push_back(getGenreText());                  // tambah kolom genre (digabung koma)
        data.push_back(sutradara);                       // tambah kolom sutradara
        return data;                                     // kembalikan 6 kolom gabungan
    }
};

File: Python/FilmBioskop.py

from Film import Film  # import class Film sebagai induk

class FilmBioskop(Film):
    """Class level 2 (turunan Film)"""

    # __init__: Inisialisasi objek FilmBioskop plus durasi, genre, dan sutradara.
    def __init__(self, id:int, judul:str, harga:int, durasi_menit:int, genre:list, sutradara:str):
        super().__init__(id, judul, harga)                 # panggil konstruktor class Film
        self.__durasi_menit = durasi_menit                 # simpan durasi tayang film (menit)
        self.__genre = genre if genre is not None else []  # simpan genre; kosong bila None
        self.__sutradara = sutradara                       # simpan nama sutradara

    # ===== getter =====
    # get_durasi_menit: Mengembalikan durasi film (menit).
    def get_durasi_menit(self)->int:
        return self.__durasi_menit                         # kembalikan durasi film (menit)

    # get_genre: Mengembalikan daftar genre film.
    def get_genre(self)->list:
        return self.__genre                                # kembalikan daftar genre

    # get_sutradara: Mengembalikan nama sutradara film.
    def get_sutradara(self)->str:
        return self.__sutradara                            # kembalikan nama sutradara

    # ===== setter =====
    # set_durasi_menit: Mengisi ulang durasi film (menit).
    def set_durasi_menit(self, durasi_menit:int)->None:
        self.__durasi_menit = durasi_menit                 # isi ulang durasi film (menit)

    # set_genre: Mengisi ulang daftar genre film.
    def set_genre(self, genre:list)->None:
        self.__genre = genre                               # isi ulang daftar genre

    # set_sutradara: Mengisi ulang nama sutradara film.
    def set_sutradara(self, sutradara:str)->None:
        self.__sutradara = sutradara                       # isi ulang nama sutradara

    # get_genre_text: Menggabungkan daftar genre menjadi satu string (dipisah koma).
    def get_genre_text(self)->str:
        """Menggabungkan daftar genre menjadi satu string (dipisah koma)."""
        return ", ".join(self.__genre)                     # gabungkan genre dengan pemisah koma

    # get_data: Mengembalikan data Film + atribut bioskop (6 kolom).
    def get_data(self)->list:
        """Mengembalikan data Film + atribut bioskop (6 kolom)."""
        data = super().get_data()                          # ambil data milik class Film
        # durasi diformat jam-menit langsung di method milik class (tanpa fungsi bantu)
        jam, sisa = divmod(self.__durasi_menit, 60)        # menit dibagi 60 menjadi jam dan sisa
        if jam > 0:                                        # sudah satu jam atau lebih?
            durasi = f"{jam} jam"                          # tampilkan dalam satuan jam
            if sisa > 0:                                   # masih ada sisa menit?
                durasi += f" {sisa} menit"                 # tambahkan sisa menit ke hasil
        else:                                              # belum genap satu jam
            durasi = f"{self.__durasi_menit} menit"        # di bawah 60 menit: tampilkan menit
        # sambungkan data Film dengan atribut khusus bioskop
        data.extend([durasi, self.get_genre_text(), self.__sutradara])
        return data                                        # kembalikan 6 kolom data film bioskop